### PR TITLE
Fixes Workspaces Bug with Fal Resource

### PR DESCRIPTION
**Pull request by Marc-Bauer**
#817

### CODE REVIEW

**Comment by xf-** [comment]
THX for your PR, please follow the CGL
Add a Prefix to commit message e.g.:
[BUGFIX] FAL viewhelper in workspaces

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 1]
```diff
@@ -25,42 +26,91 @@ class FalViewHelper extends AbstractRecordResourceViewHelper {
...
+	/**
 	 * Constructor
 	 */
 	public function __construct() {
 		$this->resourceFactory = GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Resource\\ResourceFactory');
+		$this->fileRepository = GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Resource\\FileRepository');
 	}
 
 	/**
 	 * @param mixed $identity
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: Update also PHPdoc

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 2]
```diff
@@ -25,42 +26,91 @@ class FalViewHelper extends AbstractRecordResourceViewHelper {
...
+	public function getResource($fileReference) {
 		$file = $fileReference->getOriginalFile();
 		$fileReferenceProperties = $fileReference->getProperties();
 		$fileProperties = ResourceUtility::getFileArray($file);
-
 		ArrayUtility::mergeRecursiveWithOverrule($fileProperties, $fileReferenceProperties, TRUE, FALSE, FALSE);
 		return $fileProperties;
 	}
 
+	protected function getFileReferences($table, $field,$tt_content) {
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: Missing PHPdoc

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 3]
```diff
@@ -25,42 +26,91 @@ class FalViewHelper extends AbstractRecordResourceViewHelper {
...
 	/**
 	 * @param array $record
 	 * @return array
 	 */
 	public function getResources($record) {
-		$sqlTable = $GLOBALS['TYPO3_DB']->fullQuoteStr($this->getTable(), 'sys_file_reference');
-		$sqlField = $GLOBALS['TYPO3_DB']->fullQuoteStr($this->getField(), 'sys_file_reference');
-		$sqlRecordUid = $GLOBALS['TYPO3_DB']->fullQuoteStr($record[$this->idField], 'sys_file_reference');
+		$this->contentObj = $this->configurationManager->getContentObject();
+		$this->getDatabaseConnection = $GLOBALS['TYPO3_DB'];
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: No properties set in this FalViewHelper object.
Simple variable should be enough.
Also add a PHPdoc for IDEs and better readable code.

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 4]
```diff
@@ -25,42 +26,91 @@ class FalViewHelper extends AbstractRecordResourceViewHelper {
...
+			if (!empty($references)) {
+				$referenceUids = array_keys($references);
+			}
+			if (!empty($referenceUids)) {
+				foreach ($referenceUids as $referenceUid) {
+					try {
+						// Just passing the reference uid, the factory is doing workspace
+						// overlays automatically depending on the current environment
+						$itemList[] = $this->resourceFactory->getFileReferenceObject($referenceUid);
+					} catch (ResourceDoesNotExistException $exception) {
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: Undefined Exception

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 5]
```diff
@@ -25,42 +26,91 @@ class FalViewHelper extends AbstractRecordResourceViewHelper {
...
+						// No handling, just omit the invalid reference uid
+					}
+				}
+			}
+			$images = $itemList;
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: $itemList not defined

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 6]
```diff
@@ -11,6 +11,7 @@
 use FluidTYPO3\Vhs\Utility\ResourceUtility;
 use TYPO3\CMS\Core\Utility\ArrayUtility;
 use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Utility\DebuggerUtility;
```
Comment (xf-) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php]: unnecessary import

**Comment by xf-** [comment]
I didn't test this in workspaces yet, but i talked with team and code looks OK in general.
To get it merged we need unit tests. This can break at any core change and workspaces are always more effort to test manually.
Could you create unit tests and remove the minor CGL stuff?

**Comment by Marc-Bauer** [comment]
Hi I can do it, but I'm now in holidays. The earlier moment to do that is the first of september ;)

**Comment by mneuhaus** [comment]
i've taken up this pr and updated it with cgl changes and a simple unit test over here: #939

**Pull request closed without merging**